Core/src/org/sleuthkit/autopsy/mainui/datamodel/AbstractDAO.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Autopsy Forensic Browser
+ *
+ * Copyright 2021 Basis Technology Corp.
+ * Contact: carrier <at> sleuthkit <dot> org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sleuthkit.autopsy.mainui.datamodel;
+
+import org.sleuthkit.autopsy.mainui.datamodel.events.DAOEvent;
+import java.beans.PropertyChangeEvent;
+import java.util.Collection;
+import org.sleuthkit.autopsy.mainui.datamodel.events.TreeEvent;
+
+/**
+ * Internal methods that DAOs implement.
+ */
+abstract class AbstractDAO {
+
+    /**
+     * Clear any cached data (Due to change in view).
+     */
+    abstract void clearCaches();
+
+    /**
+     * Handles an autopsy event (i.e. ingest, case, etc.). This method is
+     * responsible for clearing internal caches that are effected by the event
+     * and returning one or more DAOEvents that should be broadcasted to the
+     * views.
+     *
+     * @param evt The Autopsy event that recently came in from Ingest/Case.
+     *
+     * @return The list of DAOEvents that should be broadcasted to the views or
+     *         an empty list if the Autopsy events are irrelevant to this DAO.
+     */
+    abstract Collection<? extends DAOEvent> processEvent(PropertyChangeEvent evt);
+
+    /**
+     * Handles the ingest complete or cancelled event. Any events that are
+     * delayed or batched are flushed and returned.
+     *
+     * @return The flushed events that were delayed and batched.
+     */
+    abstract Collection<? extends DAOEvent> handleIngestComplete();
```

Comment:
We should specify the data source the ingest completed on.  There could be multiple ingests at the same time.

Reply:
Hi @bcarrier ,  it looks like the ingest completed/cancelled events just capture ingest job id.  Is that a good enough replacement or do you want me to fetch the `DataSource` for the given ingest job id?

Reply:
This is good enough for now. Let's make a story to make it more explicit as a follow on.